src/back/wgsl/writer.rs

```diff
@@ -377,17 +373,28 @@ impl<W: Write> Writer<W> {
         module: &Module,
         handle: Handle<crate::Type>,
         members: &[crate::StructMember],
+        struct_span: u32,
     ) -> BackendResult {
         write!(self.out, "struct ")?;
         self.write_struct_name(module, handle)?;
         write!(self.out, " {{")?;
         writeln!(self.out)?;
+
         for (index, member) in members.iter().enumerate() {
             // The indentation is only for readability
             write!(self.out, "{}", back::INDENT)?;
             if let Some(ref binding) = member.binding {
                 self.write_attributes(&map_binding_to_attribute(binding))?;
             }
+
+            let current_field_size = if index != members.len() - 1 {
+                let next_member = &members[index + 1];
+                next_member.offset - member.offset
+            } else {
+                struct_span - member.offset
+            };
+            write!(self.out, "@size({current_field_size}) ")?;
```

Comment:
We can avoid writing the size explicitly if it's the same as the one inferred. See example below on how to get the size:

https://github.com/gfx-rs/naga/blob/0491d39232bafc30962f6ea240ab6ed457282a62/src/back/msl/writer.rs#L3178-L3179